xarray/core/datatree_mapping.py

```diff
@@ -71,37 +71,6 @@ def check_isomorphic(
         raise TreeIsomorphismError("DataTree objects are not isomorphic:\n" + diff)
 
 
-def diff_treestructure(a: DataTree, b: DataTree, require_names_equal: bool) -> str:
```

Comment:
I moved this into `xarray/core/formatting.py` to avoid a circular dependency issue.